src/panels/config/helpers/forms/ha-schedule-form.ts

```diff
@@ -0,0 +1,383 @@
+// @ts-ignore
+import fullcalendarStyle from "@fullcalendar/common/main.css";
+import { Calendar, CalendarOptions } from "@fullcalendar/core";
+import allLocales from "@fullcalendar/core/locales-all";
+import interactionPlugin from "@fullcalendar/interaction";
+import timeGridPlugin from "@fullcalendar/timegrid";
+// @ts-ignore
+import timegridStyle from "@fullcalendar/timegrid/main.css";
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+  unsafeCSS,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { useAmPm } from "../../../../common/datetime/use_am_pm";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import "../../../../components/ha-icon-picker";
+import "../../../../components/ha-textfield";
+import {
+  getScheduleTime,
+  Schedule,
+  ScheduleDay,
+  weekdays,
+} from "../../../../data/schedule";
+import { haStyle } from "../../../../resources/styles";
+import { HomeAssistant } from "../../../../types";
+
+const defaultFullCalendarConfig: CalendarOptions = {
+  plugins: [timeGridPlugin, interactionPlugin],
+  headerToolbar: false,
+  initialView: "timeGridWeek",
+  editable: true,
+  selectable: true,
+  selectMirror: true,
+  selectOverlap: false,
+  eventOverlap: false,
+  allDaySlot: false,
+  height: "parent",
+  locales: allLocales,
+  dayHeaderFormat: { weekday: "short", month: undefined, day: undefined },
+  slotLabelFormat: { hour: "numeric", minute: undefined, meridiem: "narrow" },
+};
+
+@customElement("ha-schedule-form")
+class HaScheduleForm extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public new?: boolean;
+
+  @state() private _name!: string;
+
+  @state() private _icon!: string;
+
+  // @ts-ignore
```

Comment:
What are you ignoring

Reply:
Unused variable

Reply:
Wondering the same.  Does the change I suggested help?  Should force TS to understand `this[_${day}]` better.

Reply:
No your suggestion doesn't modify the element or class just the Schedule type

Reply:
It does yes. Thanks